samples/web/Continote/public/scripts/edit-note.js

```diff
@@ -0,0 +1,299 @@
+    // be editing right now.
+    handleNoteToEditNotFound_();
+  };
+
+  /**
+   * Handles when the "save note" button is clicked.
+   *
+   * Updates the Note that is currently being edited in the Firebase Realtime
+   * Database to use the values from the user's input.
+   *
+   * @type {!ClickEventListener}
+   * @const
+   */
+  var handleSaveNoteButtonClicked_ = function(event) {
+    event.preventDefault();
+
+    // Create a Note from the user's input.
+    var noteFromUserInput = {
+      title: pageUi_.noteTitleInput.value,
+      content: pageUi_.noteContentInput.value
+    };
+
+    // Set the value for the Note in the Firebase Realtime Database to use
+    // the Note created from the user's input.
+    setNoteInDatabase_(noteFromUserInput).then(function() {
+      // Notify the user that the Note was successfully saved.
+      pageUi_.snackbarContainer.MaterialSnackbar.showSnackbar({
+        message: "Note successfully saved!"
+      });
+    }).catch(function() {
+      // Notify the user that the Note was not saved.
+      pageUi_.snackbarContainer.MaterialSnackbar.showSnackbar({
+        message: "Note could not be saved. Please try again."
+      });
+    });
+  };
+
+  /**
+   * Handles when the "back to my notes" button is clicked.
+   *
+   * Sends the user to the My Notes page.
+   *
+   * @type {!ClickEventListener}
+   * @const
+   */
+  var handleBackToMyNotesButtonClicked_ = function(event) {
+    event.preventDefault();
+
+    // Set the browser's location to go to the My Notes page.
+    window.location.href = "/my-notes.html";
+  };
+
+  /**
+   * Handles when the Note the user wants to edit has been found.
+   *
+   * @function
+   * @param {!Note} note - The Note to edit that was found.
+   * @const
+   */
+  var handleNoteToEditFound_ = function(note) {
+    // Use the Note values to prepopulate the Note editor's inputs.
+    setNoteEditorInputsToUseValuesFromNote_(note);
+
+    // The Note was found, so show and enable only the appropriate UI
+    // elements.
+    Utils.hideAllElementsWithClassName(showOnlyWhenNoteNotFoundClass_);
+    pageUi_.noteTitleMaterialTextField.MaterialTextfield.enable();
+    pageUi_.noteContentMaterialTextField.MaterialTextfield.enable();
+    Utils.enableButtonAndAddClickListener(
+        pageUi_.saveNoteButton, handleSaveNoteButtonClicked_);
+    Utils.showAllElementsWithClassName(showOnlyWhenNoteFoundClass_);
+  };
+
+  /**
+   * Handles when the Note the user wants to edit has not been found.
+   *
+   * @function
+   * @const
+   */
+  var handleNoteToEditNotFound_ = function() {
+    // Reset the Note editor's inputs.
+    setNoteEditorInputsToUseValuesFromNote_(null);
+
+    // The Note has not been found, so show and enable only the appropriate UI
+    // elements.
+    Utils.hideAllElementsWithClassName(showOnlyWhenNoteFoundClass_);
+    pageUi_.noteTitleMaterialTextField.MaterialTextfield.disable();
```

Comment:
The Note found/not found events could occur multiple times, if the user signs in or out.

FYI: I updated the Note UI resetting code for clarity in https://github.com/firebase/firebase-continue/pull/10/commits/e91b34b5cc7eccfaaa706c7736908a958b1d06e3.

The default state should also be clear in this refactoring.
Thank you!